beangulp/testing.py

```diff
@@ -136,57 +137,80 @@ def log(msg, level=0, **kwargs):
 def _test(ctx,
           documents: List[str],
           expected: str,
-          verbose: bool, quiet: bool, exitfirst: bool):
-    """Run the importer on all DOCUMENTS and verify that it produces the desired
-    output. The desired output is stored in Beancount ledger files with a header
-    containing additional metadata.
+          verbose: int,
+          quiet: int,
+          exitfirst: bool):
+    """Test the importer.
 
-    The files are expected to be placed by default right next to the original
-    files, but with a `.beancount` suffix. You can change their location with
-    the `--expected` option.
+    Run the importer on all DOCUMENTS and verify that it produces the
+    desired output.  The desired output is stored in Beancount ledger
+    files with a header containing additional metadata and can be
+    generated with the "generate" command.
 
-    DOCUMENTS can be files or directories. Directories are walked and the
-    importer is called on all files found, except for .beancount files.
+    The name of the desired output files is derived appending a
+    ".beancount" suffix to the name of the input files, and are
+    searched in the same directory where the input document is
+    located, unless a different location is specified through the
+    "--expected DIR" option.
+
+    DOCUMENTS can be files or directories.  Directories are walked and
+    the importer is called on all files with names not ending in
+    ".beancount".  All and only the documents for which a desired
+    output file exists must be positively identify by the importer.
 
     """
-    return run_test(ctx, documents, expected, False, verbose, quiet, exitfirst)
+    return _run(ctx, documents, expected, verbose, quiet, exitfirst=exitfirst)
 
 
-@click.command('generate', short_help="Generate expected files for tests.")
+@click.command('generate')
 @click.argument('documents', nargs=-1,
                 type=click.Path(exists=True, resolve_path=True))
 @click.option('--expected', '-e', metavar='DIR',
               type=click.Path(file_okay=False, resolve_path=True),
               help="Directory containing the expecrted output files.")
+@click.option('--verbose', '-v', count=True,
+              help="Enable verbose output.")
+@click.option('--quiet', '-q', count=True,
+              help="Suppress all output.")
+@click.option('--force', '-f', is_flag=True,
```

Comment:
Sure, though honestly I don't recall a single time I've used generate without the equivalent of --force. I'll basically be using --force by default all the time.